ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/DataSourceExtractor.java

```diff
@@ -131,6 +132,35 @@ public Void visitAliasedRelation(final AliasedRelation relation, final Void ctx)
       return null;
     }
 
+    private String checkAlternatives(final SourceName sourceName) {
+      String hint = "";
+
+      final Set<SourceName> existingSources = metaStore.getAllDataSources().keySet();
```

Comment:
nit: We probably don't need this variable and can just use `metaStore.getAllDataSources().keySet()` in the `for` loop.